components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/SaveLoginDialogFragment.kt

```diff
@@ -73,23 +74,61 @@ internal class SaveLoginDialogFragment : PromptDialogFragment() {
         }
     }
 
-    private val guid by lazy { safeArguments.getString(KEY_LOGIN_GUID) }
+    private val dispatcher = dispatcher
     private val origin by lazy { safeArguments.getString(KEY_LOGIN_ORIGIN)!! }
     private val formActionOrigin by lazy { safeArguments.getString(KEY_LOGIN_FORM_ACTION_ORIGIN) }
     private val httpRealm by lazy { safeArguments.getString(KEY_LOGIN_HTTP_REALM) }
     @VisibleForTesting
     internal val icon by lazy { safeArguments.getParcelable<Bitmap>(KEY_LOGIN_ICON) }
-
     @VisibleForTesting
     internal var username by SafeArgString(KEY_LOGIN_USERNAME)
     @VisibleForTesting
     internal var password by SafeArgString(KEY_LOGIN_PASSWORD)
-
     private var validateStateUpdate: Job? = null
+    // List of logins already stored for this site.  We use this to determine
+    // how to save the new [LoginEntry] (add a new record, or update an existing record)
+    private var loginsForOrigin: Deferred<List<Login>>? = null
+
+    // Start the process to fetch `loginsForOrigin` if we need to.
+    // This makes sure that:
+    //   - We don't start the process before `feature.loginsStorage` is set
```

Comment:
nit: maybe more specifically, "before `feature` is linked to this fragment". E.g. see `PromptFeature#reattachFragment`.